drivers/abp2/abp2_saul.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_abp2
+ * @{
+ *
+ * @file
+ * @brief       ABP2 adaption to the SAUL framework.
+ *
+ * @author      David Picard <david.picard@clermont.in2p3.fr>
+ * @}
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#include "saul.h"
+#include "abp2.h"
+
+static int read_press(const void *dev, phydat_t *press)
+{
+    abp2_raw_t rawData;
+
+    if (abp2_measure_read((abp2_t *)dev, &rawData)) {
+        return -ECANCELED;
+    }
+    float p_mbar = abp2_pressure((abp2_t *)dev, &rawData);  /* pressure in mbar */
+
+    press->val[0] = p_mbar * 1000.0f;                       /* convert to ubar */
+    press->scale = -6;
+    press->unit = UNIT_BAR;
+
+    return 1;
+}
+
+static int read_temp(const void *dev, phydat_t *temp)
+{
+    abp2_raw_t rawData;
+
+    if (abp2_measure_read((abp2_t *)dev, &rawData)) {
+        return -ECANCELED;
+    }
+    float t_degc = abp2_temperature((abp2_t *)dev, &rawData);   /* temperature in Celsius */
```

Comment:
you changed abp2_temperature to return mdegC

Reply:
Fix comments and the conversion to phydat_t.